### PR TITLE
Fix SWORD targeting mobs in arrivals

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
- No longer targets mobs in arrivals area.
- Clears active target if the target enters arrivals area.
- Abilities cannot affect any mobs, turfs or objects in arrivals area.
- Re-enables the SWORD event on non-RP.

Tested semi-extensively on a number of maps with NPCs. Even if I tried to lure SWORD into arrivals it would sort of naturally move away from there (since it clears targets entering arrivals) by engaging distant targets.

It's worth mentioning that people that are in an arrivals area are _completely_ immune to all of its abilities. This has some potential for cheesing it but I think the chances of this being successful are very low due to the new targeting behavior.

## Why's this needed? 
resolves https://github.com/goonstation/goonstation/issues/5289

### CODE REVIEW

File: code/obj/critter/sword.dm [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #define SWORD_ATTACKING_RANGE 4
+#define IS_ARRIVALS(x) (istype(x, /area/shuttle/arrival/station) || istype(x, /area/station/crewquarters/cryotron) || istype(x, /area/station/hallway/secondary/oshan_arrivals))
```
Comment (BatElite) [inline code/obj/critter/sword.dm:2]: Might be good to put in a more general place? At the least it'll probably come in useful when someone gets around to fixing emagged bots getting into arrivals.

Comment (PrimeNumb) [reply code/obj/critter/sword.dm:2]: if a dev thinks it'll be useful and is Good Enough™ I could shove it into `_std/macros/area.dm` probably

Comment (Kyle2143) [reply code/obj/critter/sword.dm:2]: Is a decent idea.

**Review by virvatuli-gh — COMMENTED** [review]
Cool with this as long as it works for sure, I'll let some other devs take a look first though

File: code/modules/events/syndicate_retribution.dm [hunk 2]
```diff
@@ -5,7 +5,9 @@
 	required_elapsed_round_time = 40 MINUTES
 	weight = 88
 
+#ifdef RP_MODE
 	disabled = 1
+#endif
```
Comment (ZeWaka) [inline code/modules/events/syndicate_retribution.dm:10]:
```suggestion
 	disabled = 1
```
Still disabled for now. No dev really wants to enable it currently. I do want to see the fixes in, though

**Review by Nebulacrity — APPROVED** [review]
Looks good to me. 🙃

**Pull request merged**